src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticSourceAdapter.cs

```diff
@@ -54,6 +69,25 @@ private static Listener EnlistTarget(object target, Func<string, bool> isEnabled
             return listener;
         }
 
+        private static Listener EnlistTarget(object target, Func<string, object, object, bool> isEnabled)
```

Comment:
You can turn a `Func<string, bool>` into a `Func<string, object, object, bool>` so then you only need one 'real' code path.

Func<string, bool> original = ...;
Func<string, object, object, bool> adapter = (a, b, c) => original(a);